File: Cap02/frontend/alta-coesao/services/relatorioService.ts
import axios from 'axios'
import type { Dado } from '@/types/Dado'

export async function buscarDados(): Promise<Dado[]> {
  const { data } = await axios.get('/dados')
  return data
}

export async function exportarCsv(dados: Dado[]) {
  const csv = dados.map(d => d.valor).join(',')
  console.log(csv)
}

File: Cap02/backend/baixa-coesao/RelatorioService.cs
public class RelatorioService
{
    public byte[] GerarRelatorio(List<Dado> dados, string formato)
    {
        // regra de transformação
        var processados = dados.Select(d => d.Valor * 2).ToList();

        // decisão de formato
        if (formato == "CSV")
        {
            var csv = string.Join(",", processados);
            return Encoding.UTF8.GetBytes(csv);
        }

        if (formato == "PDF")
        {
            var pdf = $"PDF: {string.Join(",", processados)}";
            return Encoding.UTF8.GetBytes(pdf);
        }

        // log
        Console.WriteLine("Relatório gerado");

        return Array.Empty<byte>();
    }
}